integrationtests/src/test/java/io/kroxylicious/proxy/filter/RequestForwardDelayingFilter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+
+public class RequestForwardDelayingFilter implements RequestFilter {
+
+    @Override
+    public CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey, RequestHeaderData header, ApiMessage body, KrpcFilterContext filterContext) {
+        CompletableFuture<RequestFilterResult> future = new CompletableFuture<>();
+        var builder = filterContext.requestFilterResultBuilder();
+        try (var executor = Executors.newScheduledThreadPool(1)) {
```

Comment:
Hmmm maybe this is the case for per filter instance life cycles. Then again why would each filter instance need an isolated executor? Or at least one that wasn't supplied by a contributor.

Reply:
it just wants a way to delay, so I imagine running that on the netty executor would be fine. Making that available via the Contributor/Constructor would be great.

Reply:
In this test-use case, I'm happy to see the filter manage its own thread pool.     The wider point is valid: filters ought to have a way to share a thread pool.  Presumable, the share state feature we are imagining would permit this.

Reply:
True, for this test we do want to be certain it's working off the netty thread.

Reply:
Yeah, I was thinking of the wider point rather than this specific instance (which is fine).

I am expecting the shared state or the injection context API's to offer a solution something similar to the one @robobario outlined but I'm still not quite sure which or where.